C++/include/algorithm/dynamic_programming/README.md

```diff
@@ -44,3 +50,197 @@ Time               | Space
 _O(N<sup>3</sup>)_ | _O(N<sup>2</sup>)_
 
 where N is the number of matrices.
+
+## 2. Coin change
+
+Given an unlimited amount of coins with different values, find the number of ways of making change for a given value using the coins.
+
+### Input
+- #### `coin`
+  Type: `vector<int>`
+
+  Contains coins of different denomination.
+
+- #### `number_of_coins`
+  Type: `int`
+
+  Represents size of `coin` vector, or the total number of coins given.
+
+- #### `amount`
+  Type: `int`
+
+  Represents the amount for which we have to find the number of ways to make change.
+
+### Output
+- #### `table[amount][number_of_coins - 1]`
```

Comment:
I'm not sure what it is, but I feel like there must be a better way of describing the output when it's in this form.

Reply:
Should I store table[amount][number_of_coins - 1] in a variable first, and then return that variable. By doing so I can mention the variable name in the output (in the documentation).
If yes, then I have to make a change in the file coin_change.hpp. (and to similar instances like that)
example -
int number_of_ways = table[amount][number_of_coins - 1];
return number_of_ways;

Reply:
Actually I think this is fine the way that it is.